### PR TITLE
test(cli): add verdaccio for local publishing and e2e testing

### PR DESCRIPTION
**Pull request by IKatsuba**

### CODE REVIEW

File: e2e/cli-e2e/mocks/fs.mock.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { ensureDirectoryExists } from '@code-pushup/utils';
 
 // @TODO move into testing library
 export function cleanFolder<T extends object>(
-  dirName = 'tmp',
+  dirName = 'tmp/cli-e2e',
```
Comment (BioPhoton) [inline e2e/cli-e2e/mocks/fs.mock.ts:7]: Why is this here? If it avoids testing problems in the file system I suggest we go with --persist.filename instead

File: global-setup.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import { mkdir, rm } from 'fs/promises';
+import startLocalRegistry from './tools/scripts/start-local-registry';
```
Comment (BioPhoton) [inline global-setup.ts:2]: This is unused here. Should this only be imported in global.e2e?

**Review by BioPhoton — COMMENTED** [review]
Left some comments. I tried it last time and was not able to figure it out. 

Pretty cool you did! 🙌

File: global-setup.ts (hunk 2 above)
Comment (IKatsuba) [reply global-setup.ts:2]: I will remove it. Thanks

File: e2e/cli-e2e/mocks/fs.mock.ts (hunk 1 above)
Comment (IKatsuba) [reply e2e/cli-e2e/mocks/fs.mock.ts:7]: Verdaccio, by default, saves all data in the tmp directory. But when the test is starting, tmp is already removed. Changing the default dirName is avoiding it.

Comment (BioPhoton) [reply e2e/cli-e2e/mocks/fs.mock.ts:7]: > But when the test is starting, tmp is already removed. 

Could it be fixed by adopting `global-setup.ts` ? If not we should maybe document it a bit as I would not have thought about this reason.

**Review by BioPhoton — DISMISSED** [review]
Perfect!!!

**Review by BioPhoton — APPROVED** [review]
Lgtm
